File: js/scripts/scriptToNative.js
// some call saving precomputation
var diffX = mouse.x - point.x;
var diffY = mouse.y - point.y;

// modify velocity based on point decision
var distance = Math.pow(Math.pow(diffX, 2) + Math.pow(diffY, 2), 0.5);
if (distance > 20) {
    point.velocityX += diffX / distance;
    point.velocityY += diffY / distance;
} else if (distance < 15) {
    point.velocityX -= diffX / distance;
    point.velocityY -= diffY / distance;
}

// let the physics happen
stepPoint(point);

File: lua/scripts/scriptToNative.lua
-- some call saving precomputation
diffX = mouse:getX() - point:getX();
diffY = mouse:getY() - point:getY();

-- modify velocity based on point decision
distance = (diffX^2 + diffY^2)^0.5;
if distance > 20 then
	point:setVelocityX(point:getVelocityX() + (diffX) / distance);
	point:setVelocityY(point:getVelocityY() + (diffY) / distance);
elseif distance < 15 then
	point:setVelocityX(point:getVelocityX() - (diffX) / distance);
	point:setVelocityY(point:getVelocityY() - (diffY) / distance);
end

-- let the physics happen
stepPoint(point);